### PR TITLE
Terraform

### LINKED ISSUE #140: Connection dropping
**Related to #61 and working on branch `cloud-static-files`**

It seems that after a certain amount of time (maybe one hour) of the docker container running, it can no longer connect to the Azure Storage Container to read blobs. This manifests as the web app hanging whenever some data is retrieved about the data files (need to access the media container) before getting an error (see example below) in the logs. 

This may be to do with the `AZURE_URL_EXPIRATION_SECS` setting in `settings.py` (there are some minimal docs on the subject [here](https://django-storages.readthedocs.io/en/latest/backends/azure.html#azure-storage)).

It feels like the host is no longer accepting requests, or at least not responding in a way the client understands. Frustratingly, restarting the container does not always work. I _think_ destroying the database and restarting the container does work so I'm wondering if some sort of token is actually being generated in the background, which expires.

Any further suggestions from @cc-a or @dalonsoa appreciated. We would need to solve this before considering Azure blob storage as a long term solution. 

Full traceback example: 
```python
web_1  | ERROR 2022-08-03 13:38:54,207 Internal Server Error: /battDB/exps/33/
web_1  | Traceback (most recent call last):
web_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 55, in inner
web_1  |     response = get_response(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 197, in _get_response
web_1  |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 84, in view
web_1  |     return self.dispatch(request, *args, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/guardian/mixins.py", line 213, in dispatch
web_1  |     return super().dispatch(request, *args, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/base.py", line 119, in dispatch
web_1  |     return handler(request, *args, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/django/views/generic/detail.py", line 109, in get
web_1  |     context = self.get_context_data(object=self.object)
web_1  |   File "/usr/local/lib/python3.8/site-packages/django_tables2/views.py", line 217, in get_context_data
web_1  |     tables = self.get_tables()
web_1  |   File "/usr/src/app/battDB/views.py", line 366, in get_tables
web_1  |     data = self.get_tables_data()
web_1  |   File "/usr/src/app/battDB/views.py", line 393, in get_tables_data
web_1  |     if data_file.file_exists() and data_file.raw_data_file.parse:
web_1  |   File "/usr/src/app/battDB/models.py", line 518, in file_exists
web_1  |     return self.raw_data_file.exists()
web_1  |   File "/usr/src/app/common/models.py", line 328, in exists
web_1  |     return self.file.storage.exists(self.file.name)
web_1  |   File "/usr/local/lib/python3.8/site-packages/storages/backends/azure_storage.py", line 241, in exists
web_1  |     blob_client.get_blob_properties()
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/tracing/decorator.py", line 78, in wrapper_use_tracer
web_1  |     return func(*args, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_blob_client.py", line 1255, in get_blob_properties
web_1  |     blob_props = self._client.blob.get_properties(
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/tracing/decorator.py", line 78, in wrapper_use_tracer
web_1  |     return func(*args, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_generated/operations/_blob_operations.py", line 1915, in get_properties
web_1  |     pipeline_response = self._client._pipeline.run(  # type: ignore # pylint: disable=protected-access
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 211, in run
web_1  |     return first_node.send(pipeline_request)  # type: ignore
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   [Previous line repeated 2 more times]
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/policies/_redirect.py", line 158, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_shared/policies.py", line 543, in send
web_1  |     raise err
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_shared/policies.py", line 517, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   [Previous line repeated 1 more time]
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_shared/policies.py", line 313, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 71, in send
web_1  |     response = self.next.send(request)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/_base.py", line 103, in send
web_1  |     self._sender.send(request.http_request, **request.context.options),
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_shared/base_client.py", line 330, in send
web_1  |     return self._transport.send(request, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/storage/blob/_shared/base_client.py", line 330, in send
web_1  |     return self._transport.send(request, **kwargs)
web_1  |   File "/usr/local/lib/python3.8/site-packages/azure/core/pipeline/transport/_requests_basic.py", line 361, in send
web_1  |     raise error
web_1  | azure.core.exceptions.ServiceRequestError: (<urllib3.connection.HTTPSConnection object at 0x7f1a74908340>, 'Connection to liionsdenmedia.blob.core.windows.net timed out. (connect timeout=20)')
web_1  | ERROR 2022-08-03 13:38:54,285 "GET /battDB/exps/33/ HTTP/1.1" 500 211318
```

### LINKED ISSUE #170: Change who appears as "Admin"
https://github.com/ImperialCollegeLondon/Faraday-liionsden/blob/c819a987f6b162278f959accb7b432a0b9dda87f/liionsden/settings/production.py#L10

This probably should be something like `[("RSE Team", "ict-rse-team@imperial.ac.uk"), ("Dan Davies", "d.w.davies@impaerial.ac.uk")]`. Most certainly not me!

### PR DESCRIPTION
**Pull request by dandavies99**
Modifies the infrastructure setup to make use of the [terraform web app configuration guide](https://github.com/ImperialCollegeLondon/terraform_web_app_configuration), including use of Azurite as in the [django azure demo repository](https://github.com/ImperialCollegeLondon/django_azure_demo).

The main changes are to more closely match the config in the above-mentioned repositories. The ability to use a local version clone of an Azure container has also greatly simplified some of the tests.

The only kludge (a different approach to the one used in #202) is having a separate function to construct blob urls in `custom_azure.py` that checks the url, detects if the hostname is `azurite` and switches it for `localhost` if so. 

close #140 
close #170

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,7 +15,15 @@ jobs:
     needs: qa
 
     runs-on: ubuntu-latest
-
+    env:
+      ADMIN_PASSWORD: password
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+      POSTGRES_NAME: postgres
+      POSTGRES_HOST: localhost
```
Comment (cc-a) [inline .github/workflows/ci.yml:23]: Based on your `settings.py` I don't think these get used.

File: liionsden/settings/settings.py [hunk 2]
```diff
@@ -172,11 +176,9 @@
...
 # Azure blob storage for media files
 DEFAULT_FILE_STORAGE = "storages.backends.azure_storage.AzureStorage"
-AZURE_CONTAINER = "media"
-AZURE_ACCOUNT_NAME = "liionsdenmedia"
-AZURE_ACCOUNT_KEY = os.getenv("AZURE_STORAGE_ACCOUNT_KEY")
-AZURE_CUSTOM_DOMAIN = f"{AZURE_ACCOUNT_NAME}.blob.core.windows.net"
-MEDIA_URL = f"https://{AZURE_CUSTOM_DOMAIN}/{AZURE_CONTAINER}/"
+AZURE_CONTAINER = os.getenv("AZURE_STORAGE_CONTAINER")
+AZURE_CONNECTION_STRING = os.getenv("AZURE_STORAGE_CONNECTION_STRING")
+AZURE_ACCOUNT_NAME = os.getenv("AZURE_STORAGE_ACCOUNT_NAME")
```
Comment (cc-a) [inline liionsden/settings/settings.py:181]: Could get this from the connection string via parse_connection_string.

**Review by cc-a — APPROVED** [review]
Looks great. Couple of minor bits that could be removed I think.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dandavies99) [reply .github/workflows/ci.yml]: Yep good point!

File: liionsden/settings/settings.py (hunk 2 above)
Comment (dandavies99) [reply liionsden/settings/settings.py:181]: I did try this and for whatever reason it doesn't work. Replacing L181 with: `AZURE_ACCOUNT_NAME = parse_connection_string(AZURE_CONNECTION_STRING)["accountname"]`
causes
```python
#19 0.428   File "/usr/local/lib/python3.8/site-packages/azure/core/utils/_connection_string_parser.py", line 26, in parse_connection_string
#19 0.428     cs_args = [s.split("=", 1) for s in conn_str.strip().rstrip(";").split(";")]
#19 0.428 AttributeError: 'NoneType' object has no attribute 'strip'
```
During the collectstatic step when starting the container. So I'm tempted to leave this as separate for now.

**Pull request merged**